tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -0,0 +1,1434 @@
+      val paramMap = Map("blocking" -> "true".toString, "result" -> "false".toString)
+      var body: Map[String, JsValue] = Map(
+        "lifecycleEvent" -> "CREATE".toJson,
+        "triggerName" -> s"/$ns/$triggerName".toJson,
+        "authKey" -> s"${getAuthKey(wp)}".toJson)
+      body = body ++ parameters
+      val resp = requestEntity(POST, path, paramMap, Some(body.toJson.toString()))
+      val resultInvoke = new RestResult(resp.status, getRespData(resp))
+      expectedExitCode shouldBe resultInvoke.statusCode.intValue
+      if (resultInvoke.statusCode != OK) {
+        // Remove the trigger, because the feed failed to invoke.
+        this.delete(triggerName)
+      } else {
+        result
+      }
+    } getOrElse {
+      validateStatusCode(expectedExitCode, result.statusCode.intValue)
+      result
+    }
+    r
+  }
+
+  /**
+   * Fires trigger. Parameters mirror those available in the REST.
+   *
+   * @param name either a fully qualified name or a simple entity name
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def fire(name: String,
+                    parameters: Map[String, JsValue] = Map(),
+                    parameterFile: Option[String] = None,
+                    expectedExitCode: Int = OK.intValue)(implicit wp: WskProps): RestResult = {
+    val path = getNamePath(noun, name)
+    val params = parameterFile map { l =>
+      val input = FileUtils.readFileToString(new File(l))
+      input.parseJson.convertTo[Map[String, JsValue]]
+    } getOrElse parameters
+    val resp =
+      if (params.size == 0) requestEntity(POST, path)
+      else requestEntity(POST, path, body = Some(params.toJson.toString()))
+    new RestResult(resp.status.intValue, getRespData(resp))
+  }
+}
+
+class WskRestRule
+    extends RunWskRestCmd
+    with ListOrGetFromCollectionRest
+    with DeleteFromCollectionRest
+    with WaitFor
+    with BaseRule {
+
+  override protected val noun = "rules"
+
+  /**
+   * Creates rule. Parameters mirror those available in the REST.
+   *
+   * @param name either a fully qualified name or a simple entity name
+   * @param trigger must be a simple name
+   * @param action must be a simple name
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def create(name: String,
+                      trigger: String,
+                      action: String,
+                      annotations: Map[String, JsValue] = Map(),
+                      shared: Option[Boolean] = None,
+                      update: Boolean = false,
+                      expectedExitCode: Int = SUCCESS_EXIT)(implicit wp: WskProps): RestResult = {
+    val path = getNamePath(noun, name)
+    val annos = convertMapIntoKeyValue(annotations)
+    val published = shared map { s =>
```

Comment:
Here you can do `shared.getOrElse(false)`

Reply:
done